### ISSUE TITLE
How do add an onChange event to AutoComplete component?

### ISSUE THREAD

**Issue by acadgildjunior** [stackworx/formik-material-ui#192]
I need to do an API fetch whenever value of my AutoComplete select component changes. When I add onChange to AutoComplete and call my function, value of AutoComplete is not getting populated and I keep getting 'required' error and it wouldn't allow me to submit my form. How do I achieve both - be able to call my own function onChange of value and also get the value updated? This is my code:

            ```
                            <Field
                                            name="mentor"
                                            component={Autocomplete}
                                            helperText="Select Demo Mentor"
                                            options={mentorData}
                                            getOptionLabel={(option) => option.name}
                                            onChange={e => onChangeSelectHandler(e)}
                                      
                                            classes={{
                                                option: classes.option,
                                            }}
                                            renderOption={(option) => (
                                                <React.Fragment>
                                                {option.name}
                                                </React.Fragment>
                                            )}
                                            renderInput={(params) => (
                                                <MuiTextField
                                                {...params}
                                                error={touched['mentor'] && !!errors['mentor']}
                                                helperText={touched['mentor'] && errors['mentor']}
                                                label="Choose Mentor"
                                                />
                                            )}
                                            />
```

**Comment by cliedeman** [comment]
I would need to see the code from `onChangeSelectHandler` but my guess is that your handler does not include this code

```
setFieldValue(field.name, value);
```

**Comment by meir-achieved** [comment]
According to the code 
```
 onChange: onChange
      ? onChange
      : !freeSolo
      ? (
          _event: React.ChangeEvent<{}>,
          value: Value<T, Multiple, DisableClearable, FreeSolo>,
          _reason: AutocompleteChangeReason,
          _details?: AutocompleteChangeDetails<T>
        ) => {
          setFieldValue(field.name, value);
        }
      : undefined,
```
The change event handeler recives multiple parameters. The second one is the value.
Try `onChange={(event,value) => onChangeSelectHandler(event,value)}`
Or you can just pass the value to the event handler

**Comment by meirkl** [comment]
According to the code 
```
 onChange: onChange
      ? onChange
      : !freeSolo
      ? (
          _event: React.ChangeEvent<{}>,
          value: Value<T, Multiple, DisableClearable, FreeSolo>,
          _reason: AutocompleteChangeReason,
          _details?: AutocompleteChangeDetails<T>
        ) => {
          setFieldValue(field.name, value);
        }
      : undefined,
```
The change event handeler recives multiple parameters. The second one is the value.
Try `onChange={(event,value) => onChangeSelectHandler(event,value)}`
Or you can just pass the value to the event handler

**Comment by cliedeman** [comment]
Published this in formik-material-ui-lab@0.0.5 please give it a try
